Source/ablastr/fields/IntegratedGreenFunctionSolver.cpp

```diff
@@ -42,251 +38,51 @@ computePhiIGF ( amrex::MultiFab const & rho,
     domain.grow( phi.nGrowVect() ); // include guard cells
 
-    int const nx = domain.length(0);
-    int const ny = domain.length(1);
-    int const nz = domain.length(2);
-
-    // Allocate 2x wider arrays for the convolution of rho with the Green function
-    amrex::Box const realspace_box = amrex::Box(
-        {domain.smallEnd(0), domain.smallEnd(1), domain.smallEnd(2)},
-        {2*nx-1+domain.smallEnd(0), 2*ny-1+domain.smallEnd(1), 2*nz-1+domain.smallEnd(2)},
-        amrex::IntVect::TheNodeVector() );
+    // Do we grow the domain in the z-direction in the 2D mode?
+    bool const do_2d_fft = false;
 
-#if !defined(ABLASTR_USE_HEFFTE)
-    // Without distributed FFTs (i.e. without heFFTe):
-    // allocate the 2x wider array on a single box
-    amrex::BoxArray const realspace_ba = amrex::BoxArray( realspace_box );
-    // Define a distribution mapping for the global FFT, with only one box
-    amrex::DistributionMapping dm_global_fft;
-    dm_global_fft.define( realspace_ba );
-#elif defined(ABLASTR_USE_HEFFTE)
-    // With distributed FFTs (i.e. with heFFTe):
-    // Define a new distribution mapping which is decomposed purely along z
-    // and has one box per MPI rank
-    int const nprocs = amrex::ParallelDescriptor::NProcs();
-    amrex::BoxArray realspace_ba;
-    amrex::DistributionMapping dm_global_fft;
+    int nprocs = amrex::ParallelDescriptor::NProcs();
     {
-        int realspace_nx = realspace_box.length(0);
-        int realspace_ny = realspace_box.length(1);
-        int realspace_nz = realspace_box.length(2);
-        int minsize_z = realspace_nz / nprocs;
-        int nleft_z = realspace_nz - minsize_z*nprocs;
-
-        AMREX_ALWAYS_ASSERT(realspace_nz >= nprocs);
-        // We are going to split realspace_box in such a way that the first
-        // nleft boxes has minsize_z+1 nodes and the others minsize
-        // nodes. We do it this way instead of BoxArray::maxSize to make
-        // sure there are exactly nprocs boxes and there are no overlaps.
-        amrex::BoxList bl(amrex::IndexType::TheNodeType());
-        for (int iproc = 0; iproc < nprocs; ++iproc) {
-            int zlo, zhi;
-            if (iproc < nleft_z) {
-                zlo = iproc*(minsize_z+1);
-                zhi = zlo + minsize_z;
-
-            } else {
-                zlo = iproc*minsize_z + nleft_z;
-                zhi = zlo + minsize_z - 1;
-
-            }
-            amrex::Box tbx(amrex::IntVect(0,0,zlo),amrex::IntVect(realspace_nx-1,realspace_ny-1,zhi),amrex::IntVect(1));
-
-            tbx.shift(realspace_box.smallEnd());
-            bl.push_back(tbx);
-        }
-        realspace_ba.define(std::move(bl));
-        amrex::Vector<int> pmap(nprocs);
-        std::iota(pmap.begin(), pmap.end(), 0);
-        dm_global_fft.define(std::move(pmap));
+        amrex::ParmParse pp("ablastr");
+        pp.queryAdd("nprocs_igf_fft", nprocs);
+        nprocs = std::max(1,std::min(nprocs, amrex::ParallelDescriptor::NProcs()));
```

Comment:
As a follow-up PR: we could add an automated test for the case `ablastr.nprocs_igf_fft = 1` and add it in the documentation.